src/blocked.c

```diff
@@ -351,265 +420,44 @@ void serveClientsBlockedOnSortedSetKey(robj *o, readyList *rl) {
-                 * to the consumer. */
-                if (!group) {
-                    addReplyError(receiver,
-                        "-NOGROUP the consumer group this client "
-                        "was blocked on no longer exists");
-                    goto unblock_receiver;
-                } else {
-                    *gt = group->last_id;
-                }
-            }
-
-            if (streamCompareID(&s->last_id, gt) > 0) {
-                streamID start = *gt;
-                streamIncrID(&start);
-
-                /* Lookup the consumer for the group, if any. */
-                streamConsumer *consumer = NULL;
-                int noack = 0;
-
-                if (group) {
-                    noack = receiver->bpop.xread_group_noack;
-                    sds name = receiver->bpop.xread_consumer->ptr;
-                    consumer = streamLookupConsumer(group,name,SLC_DEFAULT);
-                    if (consumer == NULL) {
-                        consumer = streamCreateConsumer(group,name,rl->key,
-                                                        rl->db->id,SCC_DEFAULT);
-                        if (noack) {
-                            streamPropagateConsumerCreation(receiver,rl->key,
-                                                            receiver->bpop.xread_group,
-                                                            consumer->name);
-                        }
-                    }
-                }
-
-                /* Emit the two elements sub-array consisting of
-                 * the name of the stream and the data we
-                 * extracted from it. Wrapped in a single-item
-                 * array, since we have just one key. */
-                if (receiver->resp == 2) {
-                    addReplyArrayLen(receiver,1);
-                    addReplyArrayLen(receiver,2);
-                } else {
-                    addReplyMapLen(receiver,1);
-                }
-                addReplyBulk(receiver,rl->key);
-
-                streamPropInfo pi = {
-                    rl->key,
-                    receiver->bpop.xread_group
-                };
-                streamReplyWithRange(receiver,s,&start,NULL,
-                                     receiver->bpop.xread_count,
-                                     0, group, consumer, noack, &pi);
-                /* Note that after we unblock the client, 'gt'
-                 * and other receiver->bpop stuff are no longer
-                 * valid, so we must do the setup above before
-                 * the unblockClient call. */
-
-unblock_receiver:
-                updateStatsOnUnblock(receiver, 0, elapsedUs(replyTimer), server.stat_total_error_replies != prev_error_replies);
-                unblockClient(receiver);
-                afterCommand(receiver);
-                server.current_client = old_client;
+            robj *o = lookupKeyReadWithFlags(rl->db, rl->key, LOOKUP_NONOTIFY | LOOKUP_NOSTATS);
```

Comment:
i think it could be a good idea to also do NOTOUCH and NOEXPIRE.
the key will be touched when we run the command on it.
and i think i'd rather not do lazy expiry from this code path.

Reply:
done